### PR TITLE
Added slack username and removed irc nickname from user profile

### LINKED ISSUE #2150: Replace IRC with Slack username in Supermarket profiles
We no longer maintain an IRC channel as a community and we long ago moved to Slack. We should update the profile to allow users to set their Slack handle instead of the IRC name.

## Removal of Legacy fields:

- Profile pages like this https://supermarket.chef.io/users/tas50 do not list IRC usernames
- Profile edit page https://supermarket.chef.io/profile/edit does not include IRC username
- Existing data in the database should also be removed

## Support for Chef Community Slack Username:

- Profile pages like this https://supermarket.chef.io/users/tas50 list a Chef Community Slack username
- Profile edit page https://supermarket.chef.io/profile/edit allows you to set a Chef Community Slack username

### PR DESCRIPTION
**Pull request by RajeshPaul38**
### Description

- Removed irc_nickname from users table, user edit profile and view profile
- Added slack_username in users table, user edit profile and view profile
- Added db migrate step before running unit tests (spec)

### Issues Resolved

#2150  #2173

### Check List

- [X] New functionality includes tests
- [X] All buildkite tests pass
- [X] Full omnibus build and tests in buildkite pass
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

File: src/supermarket/app/views/profile/edit.html.erb [hunk 1]
```diff
@@ -31,9 +31,9 @@
               <%= f.label :twitter_username, "Twitter Username" %>
               <%= f.text_field :twitter_username, placeholder: 'Twitter Username', title: 'twitter username' %>
             </div>
-            <div class="ircnickname-field">
-              <%= f.label :irc_nickname, "IRC Nickname" %>
-              <%= f.text_field :irc_nickname, placeholder: 'IRC Nickname', title: 'irc nickname' %>
+            <div class="slackusername-field">
+              <%= f.label :slack_username, "Slack Username" %>
```
Comment (tas50) [inline src/supermarket/app/views/profile/edit.html.erb:35]:
```suggestion
              <%= f.label :slack_username, "Chef  Community Slack Username" %>
```

File: src/supermarket/app/views/profile/edit.html.erb [hunk 2]
```diff
@@ -31,9 +31,9 @@
...
+              <%= f.text_field :slack_username, placeholder: 'Slack Userkname', title: 'slack username' %>
```
Comment (tas50) [inline src/supermarket/app/views/profile/edit.html.erb:36]: I think there's a typo in username here

Comment (RajeshPaul38) [reply src/supermarket/app/views/profile/edit.html.erb:36]: Yeah let me change this

Comment (RajeshPaul38) [reply src/supermarket/app/views/profile/edit.html.erb]: resolved now

**Comment by RajeshPaul38** [comment]
Feature screenshots:

Screenshot 2021-08-24 at 2 51 14 AM
Screenshot 2021-08-24 at 2 51 37 AM

**Comment by tas50** [comment]
Any idea how we can get unit tests to run when we have DB migrations? @saghoshprogress @RajeshPaul38

**Comment by RajeshPaul38** [comment]
ok @tas50 let us check on this. let's keep this PR on hold for now so that we can check if the solution works and unit test is unblocked.

**Comment by RajeshPaul38** [comment]
@tas50 I've added the step to do db:migrate in test env before running the specs. Now all checks are green. It's unblocked for merging. Pls take the next step.
CC @saghoshprogress

**Pull request merged**